agent/lib_php_amqplib.c

```diff
@@ -0,0 +1,401 @@
+  zval retval;
+  int result = FAILURE;
+
+  result = zend_eval_string("class_exists('PhpAmqpLib\\Channel\\AMQPChannel');",
+                            &retval, "Get nr_php_amqplib_class_exists");
+  /*
+   * We don't need to check anything else at this point. If this fails, there's
+   * nothing else we can do anyway.
+   */
+
+  zval_dtor(&retval);
+}
+
+/*
+ * Version detection will be called directly from PhpAmqpLib\\Package::VERSION
+ * nr_php_amqplib_handle_version will automatically load the class if it isn't
+ * loaded yet and then evaluate the string. To avoid the VERY unlikely but not
+ * impossible fatal error if the file/class isn't loaded yet, we need to wrap
+ * the call in a try/catch block and make it a lambda so that we avoid fatal
+ * errors.
+ */
+void nr_php_amqplib_handle_version() {
+  char* version = NULL;
+  zval retval;
+  int result = FAILURE;
+
+  result = zend_eval_string(
+      "(function() {"
+      "     $nr_php_amqplib_version = '';"
+      "     try {"
+      "          $nr_php_amqplib_version = PhpAmqpLib\\Package::VERSION;"
+      "     } catch (Throwable $e) {"
+      "     }"
+      "     return $nr_php_amqplib_version;"
+      "})();",
+      &retval, "Get nr_php_amqplib_version");
+
+  /* See if we got a non-empty/non-null string for version. */
+  if (SUCCESS == result) {
+    if (nr_php_is_zval_non_empty_string(&retval)) {
+      version = Z_STRVAL(retval);
+    }
+  }
+
+  if (NRINI(vulnerability_management_package_detection_enabled)) {
+    /* Add php package to transaction */
+    nr_txn_add_php_package(NRPRG(txn), PHP_PACKAGE_NAME, version);
+  }
+
+  nr_txn_suggest_package_supportability_metric(NRPRG(txn), PHP_PACKAGE_NAME,
+                                               version);
+
+  zval_dtor(&retval);
+}
+
+static inline void nr_php_amqplib_get_host_and_port(
+    zval* amqp_connection,
+    nr_segment_message_params_t* message_params) {
+  zval* amqp_server = NULL;
+  zval* amqp_port = NULL;
+  zval* connect_constructor_params = NULL;
+
+  if (NULL == amqp_connection || NULL == message_params) {
+    return;
+  }
+
+  if (nr_php_is_zval_valid_object(amqp_connection)) {
+    connect_constructor_params
+        = nr_php_get_zval_object_property(amqp_connection, "construct_params");
+    if (nr_php_is_zval_valid_array(connect_constructor_params)) {
+      amqp_server
+          = nr_php_zend_hash_index_find(Z_ARRVAL_P(connect_constructor_params),
+                                        AMQP_CONSTRUCT_PARAMS_SERVER_INDEX);
+      if (nr_php_is_zval_non_empty_string(amqp_server)) {
+        message_params->server_address
+            = ENSURE_PERSISTENCE(Z_STRVAL_P(amqp_server));
+      }
+      amqp_port
+          = nr_php_zend_hash_index_find(Z_ARRVAL_P(connect_constructor_params),
+                                        AMQP_CONSTRUCT_PARAMS_PORT_INDEX);
+      if (nr_php_is_zval_valid_scalar(amqp_port)) {
+        message_params->server_port = Z_LVAL_P(amqp_port);
```

Comment:
The check above does not guarantee the value is a long.

Reply:
verify long: [095f82d](https://github.com/newrelic/newrelic-php-agent/pull/1009/commits/095f82dd24439b86044da7e882094468003b91df)

Reply:
Is it possible for this value to be a string?   Does the AWS SDK define the allowable types?

Reply:
php-amqplib only allows integers > 0.